providers/firestore.js

```diff
@@ -57,10 +57,13 @@ module.exports = class extends Provider {
 	/**
 	 * Returns all the documents of a collection
 	 * @param {string} table Name of the collection, for which all values is to be returned
+	 * @param {array} filter An optional array, to filter out the data.
 	 * @returns {Promise<Array>}
-	 */
-	getAll(table) {
-		return this.db.collection(table).get().then(snaps => snaps.docs.map(snap => snap.data()));
+	0 */
+	getAll(table, filter = []) {
```

Comment:
At this point, you should use an async function. The following lines are hard to read and follow.